BDArmory/Modules/BDDeployableRail.cs

```diff
@@ -0,0 +1,421 @@
+        {
+            get { return rdyToFire; }
+        }
+        MissileLauncher rdyMissile;
+        public MissileLauncher readyMissile
+        {
+            get { return rdyMissile; }
+        }
+
+        MissileFire wm;
+
+        public MissileFire weaponManager
+        {
+            get
+            {
+                if (wm && wm.vessel == vessel) return wm;
+                wm = VesselModuleRegistry.GetMissileFire(vessel, true);
+                return wm;
+            }
+        }
+
+        [KSPEvent(guiActive = true, guiActiveEditor = true, guiName = "toggle deployment")]//FIXME - localize later--
+        public void ToggleRail()
+        {
+            UpdateMissileChildren();
+            DeployRail(false);
+        }
+        public override void OnStart(StartState state)
+        {
+            base.OnStart(state);
+
+            part.force_activate();
+            setupComplete = false;
+            deployTransform = part.FindModelTransform(deployTransformName);
+            deployState = Misc.Misc.SetUpSingleAnimation(deployAnimName, part);
+
+            deployState.enabled = true;
+            deployState.speed = 0;
+            deployState.normalizedTime = 1;
+            deployed = true;
+
+            UpdateMissileChildren();
+
+            if (HighLogic.LoadedSceneIsFlight)
+            {
+                //DisableRail(); //In SPH, missiletransforms got, then retracting works fine. in flight, transforms got, and retracting moves it a littlebit, then deploying reveals an offset in where the transforms are. wth
+                //DeployRail(true); //this works when called manually later, but not as part of initial spawn...
+                //...but does need to occur before RCS shapshot takes place. hrm.
+                StartCoroutine(OnStartDeploy());
+            }
+            //setupComplete = true;
+        }
+
+        public void DeployRail(bool externallycalled)
+        {
+            if (!deployed) //deploy
+            {
+                StopRoutines();
+                deployRoutine = StartCoroutine(Deploy());
+            }
+            else
+            {
+                StopRoutines();
+                retractRoutine = StartCoroutine(Retract());
+            }
+            if (externallycalled) return;
+            using (List<Part>.Enumerator p = part.symmetryCounterparts.GetEnumerator())
+                while (p.MoveNext())
+                {
+                    if (p.Current == null) continue;
+                    if (p.Current != part)
+                    {
+                        p.Current.FindModuleImplementing<BDDeployableRail>().DeployRail(true);
+                    }
+                }
+        }
+        public IEnumerator OnStartDeploy()
+        {
+            yield return new WaitForSeconds(1); //figure out what the wait interval needs to be. Too soon, and the offsets get messed up. maybe have the RCS snapshot delay instead?
+            UpdateMissileChildren();
```

Comment:
have yet to look at RCSsnapshot and when it gets called/how it interacts with the rail's initial setup

Reply:
In `GetVesselRadarCrossSection`:
```csharp
// Run intensive RCS rendering if 1. It has not been done yet, 2. If the competition just started (capture vessel changes such as gear-raise or robotics)
```
The RCS snapshot gets updated whenever the vessel mass is changed by more than 2.5% after losing a part (`UpdateRCSDelayed` in TargetInfo.cs).
I think this should be fine as long as it's not deployed at the time of competition start, though that would mean that the vessel has the same stealth when the rail is deployed as when it's not.

Reply:
Yeah, I recalled it being something like that, but hadn't refreshed myself in the timing - the reason that the rails start deployed and wait a second after spawn to retract is all the various attempts I tried to have them spawn retracted resulted in missiles glitching out of position, it was only with them being retracted a moment after initialization that they worked properly. If the RCSsnapshot is getting called on comp start, not spawn, that should be fine.
Having no change in stealth cross-section on deployment is unavoidable unless the RCS snapshot is getting called every time a rail deploys - but only a single rail will deploy at any given time, so it shouldn't be that much of a difference anyway